### ISSUE TITLE
Building development system on non-NFS capable host

### ISSUE THREAD

**Issue by MBARIMike** [stoqs/stoqs#774]
Most STOQS developers use Macs; that bias affects the scripts and instructions provided for setting up a development system as [documented here](https://github.com/stoqs/stoqs#getting-started-with-a-stoqs-development-system).

The summer 2017 Pull Requests https://github.com/stoqs/stoqs/pull/594 and https://github.com/stoqs/stoqs/pull/595 modified the scripts and instructions to make it easy for a developer to use host-native tools like PyCharm for editing code. Unfortunately, because this requires NFS (Network File System) server support on the host, it doesn't work on hosts (e.g. Microsoft Windows machines) that don't support NFS.

Instructions need to be provided for MS Windows users to get set up with a STOQS development system.

**Comment by MBARIMike** [comment]
FYI @bretstine, @markmocek, @mirandasaari1... 

Here Are the changes I made to build a Vagrant development system undoing the NFS mount:

```diff
diff --git a/Vagrantfile b/Vagrantfile
index b5b0a43..265d8fb 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -19,7 +19,7 @@ Vagrant.configure("2") do |config|
   config.ssh.forward_x11 = true
   config.ssh.insert_key = false
   config.vm.network :private_network, ip: '192.168.50.50'
-  config.vm.synced_folder '.', '/vagrant', nfs: true,  mount_options: ['tcp', 'fsc' ,'actimeo=300']
+  #config.vm.synced_folder '.', '/vagrant', nfs: true,  mount_options: ['tcp', 'fsc' ,'actimeo=300']
   config.vbguest.auto_update = false
   config.vm.provision "shell", path: "provision.sh"
 end
diff --git a/docker/requirements/base.txt b/docker/requirements/base.txt
index 76e8698..2c6a689 100644
--- a/docker/requirements/base.txt
+++ b/docker/requirements/base.txt
@@ -52,7 +52,7 @@ scikit-learn==0.19.2
 scipy==1.1.0
 seawater==3.3.4
 selenium==3.14.0
-simplejson==3.16.1
+simplejson==3.16.0
 six==1.11.0
 slacker==0.9.65
 Sphinx==1.7.9
diff --git a/provision.sh b/provision.sh
index 02c457f..a62597a 100755
diff --git a/provision.sh b/provision.sh
index 02c457f..a62597a 100755
--- a/provision.sh
+++ b/provision.sh
@@ -297,8 +297,8 @@ sed -i 's#/usr/lib/openssh/sftp-server#/usr/libexec/openssh/sftp-server#' /etc/s

 echo Cloning STOQS repo from https://github.com/stoqs/stoqs.git...
 echo ">>> See CONTRIBUTING.md for how to configure your development system so that you can contribute to STOQS"
-mkdir /vagrant/dev
-cd /vagrant/dev
+mkdir /home/vagrant/dev
+cd /home/vagrant/dev
 git clone --depth=50 https://github.com/stoqs/stoqs.git stoqsgit
 cd stoqsgit
 git config core.preloadindex true
@@ -314,10 +314,10 @@ chown -R $USER /home/$USER

 echo Provisioning and setup have finished.
 echo Default database loading and STOQS software tests should be run with:
-echo "(These commands are also found in /vagrant/dev/stoqsgit/README.md)"
+echo "(These commands are also found in /home/vagrant/dev/stoqsgit/README.md)"
 echo ---------------------------------------------------------------------
 echo vagrant ssh -- -X
-echo "cd /vagrant/dev/stoqsgit && source venv-stoqs/bin/activate"
+echo "cd /home/vagrant/dev/stoqsgit && source venv-stoqs/bin/activate"
 echo export DATABASE_URL=postgis://stoqsadm:CHANGEME@127.0.0.1:5438/stoqs
 echo ./test.sh CHANGEME
```

**Comment by MBARIMike** [comment]
Hello Capstone team,

I suggest starting over with building your development systems. Read [the README](https://github.com/stoqs/stoqs) and follow those instructions.

**Comment by MBARIMike** [comment]
It seems that there are still problems in the instructions for installation on a Windows 10 system.

**Comment by MBARIMike** [comment]
When building on a WIndows 10 system this error happens:
```
    default: Modifying local firewall to allow incoming connections on ports 80 and 8000
    default: FirewallD is not running
    default: FirewallD is not running
    default: FirewallD is not running
    default: Configuring vim edit environment
    default: Configure and restart sshd for enabling PyCharm interpreter
    default: Cloning STOQS repo from https://github.com/stoqs/stoqs.git into /home/vagrant/dev...
    default: >>> See CONTRIBUTING.md for how to configure your development system so that you can contribute to STOQS
    default: Cloning into 'stoqsgit'...
    default: fatal: unable to access 'https://github.com/stoqs/stoqs.git/': Could not resolve host: github.com; Unknown error
```
The key error is `Could not resolve host: github.com`. Perhaps the network is getting scroggled by one of the steps in `provision.sh` just before the `git clone` is executed?

**Comment by MBARIMike** [comment]
Testing whether this [network restart line](https://github.com/stoqs/stoqs/blob/943a3e1d6ab61589446fed927b438358ee26abcb/provision.sh#L217) in provision.sh is causing the VM's network on a Windows host to break before `git clone` is executed...

**Comment by MBARIMike** [comment]
That seems to have been the problem. Interesting that it doesn't seem to be a problem on MacOS hosts.

This [commit](https://github.com/stoqs/stoqs/pull/792/commits/6a6cbc8d7b901d5519aa37646d54212e0d4a1ecc) should fix the problem.

**Comment by MBARIMike** [comment]
With the [merged PR](https://github.com/stoqs/stoqs/pull/792) I was able to build a development system on Windows 10 by following the steps in [README.md](https://github.com/stoqs/stoqs/blob/d1ee577b221b1a4c9eb517cce0103601b3a74f2c/README.md). I first removed my previous failed builds by doing a `vagrant destroy` in the Vagrants\stoqsvm directory and then removing the directory and all its contents. After curl'ing the provision.sh and Vagrantfile files I edited them to disable the NFS mount. The build took just over an hour. The Unit tests ran OK: 

```
Unit tests...
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
........................................
----------------------------------------------------------------------
Ran 40 tests in 53.444s

OK
```
But the Functional tests did not display the Firefox windows as they should. I suspect an issue with X-Windows - even though I have Xming installed, I don't think the X11 forwarding is working with the `-X` option of the `vagrant ssh` command. I was able to start the development server and see the Initial test database at http://localhost:8008/default.
